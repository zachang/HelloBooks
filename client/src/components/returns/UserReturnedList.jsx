import React from 'react';
import PropTypes from 'react-proptypes';
import { Link } from 'react-router';

/**
 * @function UserReturnedList
 * @param {object} props
 * @return {XML} JSX
 */
const UserReturnedList = props => (
  <tr>
    <td>{ props.returner.Book.bookName }</td>
    <td>{ props.returner.User.fullname}</td>
    <td>{ new Date(props.returner.borrowDate).toLocaleString()}</td>
    <td>{ (props.returner.expectedReturn === null) ? 'None' : new Date(props.returner.expectedReturn).toLocaleString() }</td>
    <td>{ new Date(props.returner.actualReturn).toLocaleString() }</td>
    <td>No</td>
    <td>
      { (props.returner.returned === 'true' || (props.clickedReturnedBorrowList.indexOf(props.returner.id) > -1)) ? 'Confirmed' :
        <Link
          className='waves-effect waves-light btn btn-small'
          onClick={() => (props.confirmReturn(props.returner.id))}
        >
          Confirm
        </Link>
      }
    </td>
  </tr>
);

UserReturnedList.propTypes = {
  returner: PropTypes.object.isRequired,
  clickedReturnedBorrowList: PropTypes.array.isRequired,
  confirmReturn: PropTypes.func.isRequired,
};
export default UserReturnedList;