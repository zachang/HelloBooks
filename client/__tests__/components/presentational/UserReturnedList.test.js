import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import UserReturnedList from '../../../src/components/returns/UserReturnedList.jsx';

describe('<UserReturnedList/>', () => {
  const props = {
    returner: {
      Book: {
        bookName: 'Science',
        borrowDate: new Date(),
        expectedReturn: new Date(),
        actualReturn: 'Science',
        id: 1,
      },
      User: {
        fullname: 'Tito Navo'
      }
    },
    confirmReturn: () => '',
    clickedReturnedBorrowList: []
  };

  const shallowWrapper = shallow(<UserReturnedList {...props}/>);
  const tree = toJson(shallowWrapper);
  let wrapper;

  const wrapperMount = mount(<UserReturnedList {...props} />);

  it('renders <UserReturnedList /> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick', () => {
    wrapper = shallow(<UserReturnedList {...props} />);
    wrapper.find('.btn-small').simulate('click');
    expect(wrapperMount.find('.btn-small').at(0).text()).toBe('Confirm');
  });
});

