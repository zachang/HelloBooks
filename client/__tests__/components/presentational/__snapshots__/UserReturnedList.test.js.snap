// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserReturnedList/> renders <UserReturnedList /> component 1`] = `
<tr>
  <td>
    Science
  </td>
  <td>
    Tito Navo
  </td>
  <td>
    Invalid Date
  </td>
  <td>
    Invalid Date
  </td>
  <td>
    Invalid Date
  </td>
  <td>
    No
  </td>
  <td>
    <Link
      className="waves-effect waves-light btn btn-small"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
    >
      Confirm
    </Link>
  </td>
</tr>
`;
